provisioning/service/src/HttpDeviceRegistrationClient.cs

```diff
@@ -0,0 +1,660 @@
+                }
+            };
+
+            return this.httpClientHelper.DeleteAsync<IETagHolder>(GetEnrollmentUri(registrationId), eTagHolder, errorMappingOverrides, null, cancellationToken);
+        }
+
+        public override Task RemoveEnrollmentsAsync(IEnumerable<Enrollment> enrollments)
+        {
+            return RemoveEnrollmentsAsync(enrollments, CancellationToken.None);
+        }
+
+        public override Task RemoveEnrollmentsAsync(IEnumerable<Enrollment> enrollments, CancellationToken cancellationToken)
+        {
+            this.EnsureInstanceNotClosed();
+
+            foreach (Enrollment enrollment in enrollments)
+            {
+                if (string.IsNullOrWhiteSpace(enrollment.ETag))
+                {
+                    throw new ArgumentException(ApiResources.ETagNotSetWhileDeletingEnrollment);
+                }
+            }
+
+            var operation = new BulkOperation()
+            {
+                Mode = BulkOperationMode.Delete,
+                Enrollments = enrollments
+            };
+
+            return this.httpClientHelper.PostAsync<BulkOperation, BulkOperationResult>(GetBulkEnrollmentUri(), operation,
+                new Dictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>(), null, cancellationToken);
+        }
+
+        public override Task RemoveEnrollmentsAsync(IEnumerable<string> registrationIds)
+        {
+            return this.RemoveEnrollmentsAsync(registrationIds, CancellationToken.None);
+        }
+
+        public override Task RemoveEnrollmentsAsync(IEnumerable<string> registrationIds, CancellationToken cancellationToken)
+        {
+            var enrollments = new List<Enrollment>();
+            foreach (string registrationId in registrationIds)
+            {
+                enrollments.Add(new Enrollment(registrationId)
+                {
+                    ETag = "*"
+                });
+            }
+
+            return this.RemoveEnrollmentsAsync(enrollments, cancellationToken);
+        }
+
+        public override Task<Enrollment> UpdateEnrollmentAsync(Enrollment enrollment)
+        {
+            return this.UpdateEnrollmentAsync(enrollment, CancellationToken.None);
+        }
+
+        public override Task<Enrollment> UpdateEnrollmentAsync(Enrollment enrollment, CancellationToken cancellationToken)
+        {
+            return this.UpdateEnrollmentAsync(enrollment, false, cancellationToken);
+        }
+
+        public override Task<Enrollment> UpdateEnrollmentAsync(Enrollment enrollment, bool forceUpdate)
+        {
+            return this.UpdateEnrollmentAsync(enrollment, forceUpdate, CancellationToken.None);
+        }
+
+        public override Task<Enrollment> UpdateEnrollmentAsync(Enrollment enrollment, bool forceUpdate, CancellationToken cancellationToken)
+        {
+            this.EnsureInstanceNotClosed();
```

Comment:
We should remove all `this.` following the coding guideline.
Current code isn't consistent either (see below).